pkg/informer/informer.go

```diff
@@ -87,14 +104,40 @@ type DynamicDiscoverySharedInformerFactory struct {
 	subscribers     map[string]chan<- struct{}
 }
 
-// NewDynamicDiscoverySharedInformerFactory returns a factory for shared
-// informers that discovers new types and informs on updates to resources of
+// NewDynamicDiscoverySharedInformerFactory returns a factory for cluster-aware
+// shared informers that discovers new types and informs on updates to resources of
 // those types.
+// It retrieves GVR-related information from CRD informers.
 func NewDynamicDiscoverySharedInformerFactory(
 	cfg *rest.Config,
 	filterFunc func(obj interface{}) bool,
 	crdInformer kcpapiextensionsv1informers.CustomResourceDefinitionClusterInformer,
 	indexers cache.Indexers,
+) (*DynamicDiscoverySharedInformerFactory, error) {
+	crdIndexer := crdInformer.Informer().GetIndexer()
+
+	// Add an index function that indexes a CRD by its group/version/resource.
```

Comment:
How is this unique to the cluster-aware case?

Reply:
Seems like this should be closed over by the crd source - that's where the index is used and must be set

Reply:
Fixed